io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/AnalyseXLSSchemaTable.java

```diff
@@ -67,12 +71,53 @@ protected void headerCell(int num, String text) {
 	 *      org.apache.poi.ss.usermodel.Row)
 	 */
 	@Override
-	protected void analyseRow(int num, Row row) {
+	protected void analyseRow(int num, Row row, Sheet sheet) {
 		List<String> rowContent = new ArrayList<String>();
 		for (int i = 0; i < row.getLastCellNum(); i++) {
-			rowContent.add(extractText(row.getCell(i)));
+
+			Cell cell = row.getCell(i);
+			// Check if the cell is part of a merged region
+			if (cell != null && isCellPartOfMergedRegion(cell, sheet)) {
+				// Get the merged region
+				CellRangeAddress mergedRegion = getMergedRegion(cell, sheet);
+
+				// Get the first cell of the merged region (top-left cell)
+				Row mergedRow = sheet.getRow(mergedRegion.getFirstRow());
+				Cell mergedCell = mergedRow.getCell(mergedRegion.getFirstColumn());
+
+				// Retrieve the value from the merged cell
+				rowContent.add(mergedCell.getStringCellValue());
```

Comment:
If merged cells exist that contain formulas, numbers, etc., this will fail.

To cover merged cells in a more general way, I'd suggest the following approach:

- move the `isCellPartOfMergedRegion` check to the `XLSUtil.extractText` method
- if the cell is part of a merged region, apply the rest of the `extractText` logic to the upper left cell of the region

Reply:
Done! thank you for the feedback